### PR TITLE
Add percentage drain

### PR DESCRIPTION
**Pull request by Ekscentricitet**
Should close https://github.com/BartoszCichecki/LenovoLegionToolkit/issues/1077. Implements percentage drainage calculation per hour. I also added an option to calculate it based on remaining or total charge because I couldn't decide which one is better.

Default:

image

Setting:

image

Toggled:

image

## Summary by CodeRabbit

- **New Features**
	- Added an option to calculate battery discharge percentage based on remaining charge or full charge capacity.
	- Introduced a new toggle in settings for managing the discharge percentage calculation preference.

- **Enhancements**
	- Updated user interface elements and text values across various settings and information pages for clarity and additional functionality.

### CODE REVIEW

**Comment by Ekscentricitet** [comment]
Note: This will probably have merge conflicts with https://github.com/BartoszCichecki/LenovoLegionToolkit/pull/1171.

File: LenovoLegionToolkit.WPF/Resources/Resource.resx [hunk 1]
```diff
@@ -1988,48 +1988,54 @@ Supported formats are: {1}.</value>
...
   </data>
   <data name="AbstractAutomationStepControl_Delete" xml:space="preserve">
-	  <value>Delete</value>
+    <value>Delete</value>
+  </data>
+  <data name="SettingsPage_DischargePercentageBasedOnRemainingCharge_Message" xml:space="preserve">
+    <value>When toggled, the discharge rate in percentage will be calculated based on the remaining charge. Otherwise, it will be calculated based on the full charg capacity.</value>
+  </data>
+  <data name="SettingsPage_DischargePercentageBasedOnRemainingCharge_Title" xml:space="preserve">
+    <value>Discharge percentage based on remaining charge</value>
```
[the remark this answers is not in the document]
Comment (Ekscentricitet) [reply LenovoLegionToolkit.WPF/Resources/Resource.resx:2039]: Can't argue about descriptions with the LM.

File: LenovoLegionToolkit.WPF/Pages/SettingsPage.xaml.cs [hunk 2]
```diff
@@ -496,6 +498,19 @@
...
+        if (_isRefreshing)
+            return;
+
+        var state = _dischargePercentageBasedOnRemainingChargeToggle.IsChecked;
+        if (state is null)
+            return;
+
+        _settings.Store.IsDischargePercentageBasedOnRemainingCharge = state.Value;
+        _settings.SynchronizeStore();
+    }
```
[the remark this answers is not in the document]
Comment (BartoszCichecki) [reply LenovoLegionToolkit.WPF/Pages/SettingsPage.xaml.cs:527]: Not needed here.

**Review by BartoszCichecki — COMMENTED** [review]
First of all apologies, for the delay - busy weeks recently.

To me the only reasonable number here is the percentage drain  of full charge capacity per hour. So for 80Wh full charge capacity (**not** design capacity) and discharge rate of -20W the number would be -25%. But overall this unit of measurement doesn't make much sense. It could maybe be -25%/h, but this also isn't accurate, because it isn't really "per hour".

I am hesitant to merge this, because I can't see any value (and should probably close the issue as well) in showing a number like this. What value does it bring in the end?

**Comment by Ekscentricitet** [comment]
No problem, thanks for still being involved in the project!

Well, such measurement is not going to be accurate for forecasts. It is more of a momentary state that changes second to second. The only real use I see is if somebody wonders what theoretical percentage of the battery would have been charged if a battery could be charged at a constant rate, which can be calculated by the users themselves. It also clutters the UI and to make it make sense with further explanation, you will clutter it even more. The only reason why I implemented it was because it is simple and it would close the issue that I already made a commit for. I don't mind if you decide to close the merge request.

**Pull request closed without merging**